### PR TITLE
multi upload converter and implementation in cheque memo

### PR DESCRIPTION
**Pull request by ashishbeehyv**
## Summary by CodeRabbit

- **New Features**
	- Introduced a custom hook to combine multiple PDF files into a single document.
	- Enhanced document handling in the case management workflow, allowing multiple file uploads for cheque submissions and other document types.
	- Expanded accepted file formats to include PNG alongside JPG and PDF for various upload components.

- **Bug Fixes**
	- Improved logic for processing document uploads, ensuring better validation and handling of combined files.
	- Enhanced user feedback mechanisms for error handling and submission confirmations.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useCombineMultipleFiles.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      type: response.data.type,
+    });
+    return file;
+  } catch (error) {
+    console.error("Error fetching file:", error);
+    throw error;
+  }
+};
+
+const useCombineMultipleFiles = async (pdfFilesArray, finalFinalName = "combined-document.pdf") => {
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useCombineMultipleFiles.js:20]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useCombineMultipleFiles.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      const uri = `${window.location.origin}${Urls.FileFetchById}?tenantId=${tenantId}&fileStoreId=${fileStore}`;
+      debugger;
+      const draftFile = await getFileByFileStoreId(uri);
+      debugger;
+      formData.append("documents", draftFile);
+    } else formData.append("documents", file);
+  }
+
+  try {
+    const response = await axios.post("https://dristi-kerala-dev.pucar.org/egov-pdf/dristi-pdf/combine-documents", formData, {
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/hooks/dristi/useCombineMultipleFiles.js:36]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 3]
```diff
@@ -137,3 +138,50 @@ export const extractFeeMedium = (feeName) => {
...
+
+export const combineMultipleFiles = async (pdfFilesArray, finalFileName = "combined-document.pdf") => {
+  const tenantId = Digit.ULBService.getCurrentTenantId();
+  const formData = new FormData();
+
+  for (const file of pdfFilesArray) {
+    const { fileStore } = file;
+    if (fileStore) {
+      // ${Urls.FileFetchById} // check-Should use this but it is causing circular dependency, need to relocate Urls
+      const uri = `${window.location.origin}/filestore/v1/files/id?tenantId=${tenantId}&fileStoreId=${fileStore}`;
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:166]: will resolve later as the Url object's location change will have to be done carefully as it is used at many places

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 4]
```diff
@@ -137,3 +138,50 @@ export const extractFeeMedium = (feeName) => {
...
+      const draftFile = await getFileByFileStoreId(uri);
+      formData.append("documents", draftFile);
+    } else formData.append("documents", file);
+  }
+
+  try {
+    // ${Urls.CombineDocuments} // check- Should use this but it is causing circular dependency, need to relocate Urls
+    const combineDocumentsUrl = `${window.location.origin}/egov-pdf/dristi-pdf/combine-documents`;
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:174]: will resolve later as the Url object's location change will have to be done carefully as it is used at many places

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 5]
```diff
@@ -137,3 +138,50 @@ export const extractFeeMedium = (feeName) => {
...
+
+  for (const file of pdfFilesArray) {
+    const { fileStore } = file;
+    if (fileStore) {
+      // ${Urls.FileFetchById} // check-Should use this but it is causing circular dependency, need to relocate Urls
+      const uri = `${window.location.origin}/filestore/v1/files/id?tenantId=${tenantId}&fileStoreId=${fileStore}`;
+      const draftFile = await getFileByFileStoreId(uri);
+      formData.append("documents", draftFile);
+    } else formData.append("documents", file);
+  }
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:170]: done

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 6]
```diff
@@ -137,3 +138,50 @@ export const extractFeeMedium = (feeName) => {
...
+    // ${Urls.CombineDocuments} // check- Should use this but it is causing circular dependency, need to relocate Urls
+    const combineDocumentsUrl = `${window.location.origin}/egov-pdf/dristi-pdf/combine-documents`;
+    const response = await axios.post(combineDocumentsUrl, formData, {
+      headers: {
+        "Content-Type": "multipart/form-data",
+      },
+      responseType: "blob", // To handle the response as a Blob
+    });
+    const file = new File([response.data], finalFileName, { type: response.data.type });
+    return [file];
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:182]: no, we require it inside array

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js [hunk 7]
```diff
@@ -137,3 +138,50 @@ export const extractFeeMedium = (feeName) => {
...
+  } catch (error) {
+    console.error("Error:", error);
+    throw error;
+  }
+};
```
[the remark this answers is not in the document]
Comment (ashishbeehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/Utils/index.js:187]: done

**Review by anirudh-0 — APPROVED** [review]

**Pull request merged**